From fe739c8f8a111ed57f9595e254c0a63d92422f11 Mon Sep 17 00:00:00 2001
From: Misael Lopez Cruz <misael.lopez@ti.com>
Date: Wed, 23 Feb 2011 22:32:08 -0600
Subject: [PATCH 41/60] ASoC: McPDM: Balance pm_runtime counter for invalid stream

commit d6e374234caca1dd9f0113b387d41e7530b400f4 upstream

When playback stream is not started (e.g. parameters not supported,
file doesn't exist, etc), pm_runtime resource is never acquired as
hw_params for McPDM never gets executed. In that scenario, during
stream shutdown pm_runtime resource should not be released otherwise
it unbalances usage counter.

Change-Id: I2da9931f5e5adf3656218de3baabb9c4a1982382
Signed-off-by: Misael Lopez Cruz <misael.lopez@ti.com>
Integrated-by: Jingdong Lu <jingdong.lu@windriver.com>
---
 sound/soc/omap/omap-mcpdm.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sound/soc/omap/omap-mcpdm.c b/sound/soc/omap/omap-mcpdm.c
index 3eff7d6..0dba7f0 100644
--- a/sound/soc/omap/omap-mcpdm.c
+++ b/sound/soc/omap/omap-mcpdm.c
@@ -737,8 +737,8 @@ static void playback_abe_work(struct work_struct *work)
 		omap_mcpdm_stop(mcpdm, SNDRV_PCM_STREAM_PLAYBACK);
 		omap_mcpdm_playback_close(mcpdm, mcpdm->downlink);
 		abe_dsp_mcpdm_shutdown();
+		abe_dsp_pm_put();
 	}
-	abe_dsp_pm_put();
 	mutex_unlock(&mcpdm->mutex);
 
 	if (!mcpdm->free && !mcpdm->ul_active)
-- 
1.7.4.1

